polus-graph-pyramid-builder-plugin/src/main.py

```diff
@@ -62,46 +70,30 @@ def load_csv(fpath):
     # Load the data
     if is_coded:
         data = pandas.read_csv(fpath,skiprows=[1],usecols=[c[0] for c in cnames])
+
     else:
         data = pandas.read_csv(fpath,usecols=[c[0] for c in cnames])
 
     return data, cnames
 
-def bin_data(data,column_names):
-    """ Bin the data
-    
-    Data from a pandas Dataframe is binned in two dimensions. Binning is performed by
-    binning data in one column along one axis and another column is binned along the
-    other axis. All combinations of columns are binned without repeats or transposition.
-    There are only 20 bins in each dimension, and each bin is 1/20th the size of the
-    difference between the maximum and minimum of each column.
-
-    Inputs:
-        data - A pandas Dataframe, with nfeats number of columns
-        column_names - Names of Dataframe columns
-
-    Outputs:
-        bins - A numpy matrix that has shape (nfeats,nfeats,200,200)
-        bin_feats - A list containing the minimum and maximum values of each column
-        linear_index - Numeric value of column index from original csv
+def binning_data(data, yaxis, typegraph, column_bin_size, bin_stats):
+    """ This function bins the data """
```

Comment:
Unless there is another function called `bin_data`, the name of this function should be reverted to `bin_data`.

This also needs a complete description.

The `yaxis` argument is not used to calculate anything. Is there a reason this is an input argument and not something that is initialized as an output from within the method?

The `bin_data` argument seems relatively unnecessary since it can be calculated within the method. It seems like in the case of log scaled data it is calculated, but in the case of linear scaled data, it is expected to be precomputed.

Reply:
I changed it to bin_data. I had some variables that I named bin_data, but I changed the variables to binned_data now. 

So I added functions transform_data_linear and transforrm_data_log, that have better descriptions.
Is there something specific that I should include in the description. 

I will try to calculate to the linear y-axis outside the method.